plugins/org.eclipse.mat.parser/src/org/eclipse/mat/parser/internal/SnapshotFactoryImpl.java

```diff
@@ -196,6 +198,8 @@ public ISnapshot openSnapshot(File file, Map<String, String> args, IProgressList
                 prefix += snapshot_identifier + "."; //$NON-NLS-1$
             }
 
+            wrappedListener.setFile(new File(prefix + "log.index"));

```

Comment:
Note, seems like if anything throws before prefix setup, it will never setFile and make it to the output file. I think that's ok but maybe confusing for some operator? I don't have any suggestions on how to do it better, only an observation.

Reply:
Yeah, I wasn't sure what to do about that either, but the message will still go to the underlying `IProgressListener`, so for Eclipse that'll be the `Error Log` and for headless, that'll be `System.out`, so only the data in `*.log.index` is lost but usually that'll be because we can't write to it anyway